extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/boot/registry/PreconfiguredServiceRegistryBuilder.java

```diff
@@ -106,7 +106,13 @@ public StandardServiceRegistryImpl buildNewServiceRegistry() {
         final Map settingsCopy = new HashMap();
         settingsCopy.putAll(configurationValues);
 
-        destroyedRegistry.resetAndReactivate(bootstrapServiceRegistry, initiators, providedServices, settingsCopy);
+        // FIXME: resetAndReactivate() throws "IllegalStateException: Can't reactivate an active registry!"
+        //  during persistenceProvider.generateSchema() execution (a new PersistenceProvider instance is constructed
+        //  during this call).
+        //  Is it OK to skip the resetAndReactivate() call when the registry is already active?
+        if (!destroyedRegistry.isActive()) {
```

Comment:
this looks also a bit tricky, it should be possible to get ORM to generate the scripts without having to go through the bootstrap process of the registry.

I should probably try to find some time to help with this.

Reply:
That would be great if you could look at that. I don't really have any insight into that code, I would most likely break it if I tried to refactor that bootstrap code :). Thanks Sanne!

Reply:
I pushed a commit to change how we retrieve the information: instead of being pulled at runtime, it's now generated on bootstrap. This works around all the problems related to generation of the schema after bootstrap.